forest/intake_loader.py

```diff
@@ -0,0 +1,395 @@
+        zarr_kwargs={'consolidated': True, 'decode_times': False},
+        cdf_kwargs={'chunks': {}, 'decode_times': False})
+    try:
+        parent_source_id = [l1 for l1 in dset_dict.values()][0].parent_source_id
+    except IndexError:
+        import pdb
+        pdb.set_trace()
+    ds_label = f'{activity_id}.{institution_id}.{parent_source_id}.{experiment_id}.{table_id}.{grid_label}'
+    xr = dset_dict[ds_label]
+    print(xr[variable_id])
+    cube = xr[variable_id].to_iris()
+
+    coord_names = [c1.name() for c1 in cube.coords()]
+    if 'air_pressure' in coord_names:
+        cube.coord('air_pressure').convert_units('hPa')
+    return cube[0]  # drop member dimension
+
+
+class IntakeView(object):
+    """
+    View object for CMIP6 dataset. The main customisation of the standard view
+    class is to provide more CMIP6 specific info in the hover info.
+    """
+    def __init__(self, loader, color_mapper):
+        self.loader = loader
+        self.color_mapper = color_mapper
+        self.source = bokeh.models.ColumnDataSource({
+            "x": [],
+            "y": [],
+            "dw": [],
+            "dh": [],
+            "image": []})
+
+    def render(self, state):
+        self.source.data = self.loader.image(state)
+
+    def add_figure(self, figure):
+        renderer = figure.image(
+            x="x",
+            y="y",
+            dw="dw",
+            dh="dh",
+            image="image",
+            source=self.source,
+            color_mapper=self.color_mapper)
+        tool = bokeh.models.HoverTool(
+            renderers=[renderer],
+            tooltips=[
+                ("Name", "@name"),
+                ("Value", "@image @units"),
+                ('Valid', '@valid{%F %H:%M}'),
+                ("Level", "@level"),
+                ("Experiment", "@experiment"),
+                ("Institution", "@institution"),
+                ("Member", "@memberid"),
+                ('Variable', "@variableid"),
+            ],
+            formatters={
+                'valid': 'datetime',
+            })
+        figure.add_tools(tool)
+        return renderer
+
+
+@functools.lru_cache(maxsize=16)
+def _get_bokeh_image(cube,
+                     experiment_id,
+                     variable_id,
+                     institution_id,
+                     initial_time,
+                     member_id,
+                     selected_time,
+                     pressure,
+                     ):
+    """
+    A helper function to do  the creation of the image dict required by bokeh.
+    This includes downloading the actual data required for the current view, so
+    this function is cached to reduce remote queries.
+    """
+
+    def time_comp(select_time, time_cell):  #
+        data_time = gridded_forecast._to_datetime(time_cell.point)
+        try:
+            if abs((select_time - data_time).days) < 2:
+                return True
+        except ValueError:
+            pass
```

Comment:
Sorry - why is this exception handler necessary?

Reply:
Oh dear, more debug code :( Removed in 13fcba0.